src/virt.c

```diff
@@ -338,6 +350,29 @@ lv_config (const char *key, const char *value)
         if (ignorelist_add (il_block_devices, value)) return 1;
         return 0;
     }
+
+    if (strcasecmp (key, "BlockDeviceFormat") == 0) {
+        if (strcasecmp (value, "dev") == 0)
+            blockdevice_format = dev;
+        else if (strcasecmp (value, "source") == 0)
+            blockdevice_format = source;
+        else {
+            ERROR (PLUGIN_NAME " plugin: unknown BlockDeviceFormat: %s", value);
+            return -1;
+        }
+        return 0;
+    }
+    if (strcasecmp (key, "BlockDeviceFormatBasename") == 0) {
+        if (strcasecmp (value, "true") == 0)
+            blockdevice_format_basename = "true";
+        else if (strcasecmp (value, "false") == 0)
+            blockdevice_format_basename = "false";
+        else {
```

Comment:
Please mimic the style of the rest of the file, e.g. curly braces on their own line.